src/core/layout/qgslayoutitempicture.cpp

```diff
@@ -92,6 +92,9 @@ void QgsLayoutItemPicture::draw( QgsLayoutItemRenderContext &context )
   // painter is scaled to dots, so scale back to layout units
   painter->scale( context.renderContext().scaleFactor(), context.renderContext().scaleFactor() );
 
+  const bool prevSmoothTransform = painter->testRenderHint( QPainter::RenderHint::SmoothPixmapTransform );
+  painter->setRenderHint( QPainter::RenderHint::SmoothPixmapTransform, true );
```

Comment:
I think this should be dependent on whether the antialiasing option is present in the layout context. (For users it's a very similar thing -- "smoothed edges" vs "pixelated".